core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,497 @@
+    evaluated_both_rows['stddev'] = (((n_old - 1) * stddev_old ** 2 + (n_new - 1) * stddev_new ** 2) / (n_old + n_new - 1) +
+                                     (n_old * n_new * (mean_old - mean_new) ** 2) / ((n_old + n_new) * (n_old + n_new - 1))) ** 0.5
+    evaluated_both_rows['p_value'] = None
+    evaluated_both_rows['n'] = n_old + n_new
+    # remove features evaluated in df_new from using_prev_fit_fi if they exist
+    using_prev_fit_fi.difference_update(evaluated_new_rows.index.tolist())
+    result = pd.concat([evaluated_both_rows, evaluated_new_only_rows, evaluated_old_only_rows, evaluated_neither_rows]).sort_values('importance')
+    assert len(result) == len(df_new), "Length of the updated DataFrame must be equal to the inputted DataFrame."
+    return result
+
+
+def sort_features_by_priority(features: List[str], prioritized: Set[str], prev_importance_df: pd.DataFrame, using_prev_fit_fi: Set[str]) -> List[str]:
+    """
+    Return a list of features sorted by feature importance calculation priority in ascending order.
+    If prev_importance_df does not exist and not using auto_threshold, return the original list.
+    If prev_importance_df exists, features whose importance scores have never been calculated are
+    prioritized first followed by features whose importance scores are from previous fitted models and
+    lastly the rest of the features sorted by previous importance scores estimates in ascending order. If using
+    auto_threshold mode, noise columns are prioritized first since their scores are needed to determine
+    the pruning threshold.
+    """
+    is_first_run = prev_importance_df is None
+    auto_threshold = len(prioritized) > 0
+    if not is_first_run:
+        prev_deleted_features = [feature for feature in prev_importance_df.index if feature not in features]
+        prev_importance_df = prev_importance_df.drop(prev_deleted_features)
+        unevaluated_rows = prev_importance_df[prev_importance_df['importance'].isna()]
+        prev_fit_evaluated_rows = prev_importance_df[~(prev_importance_df['importance'].isna()) &
+                                                      (prev_importance_df.index.isin(using_prev_fit_fi))].sort_values(by='importance')
+        curr_fit_evaluated_rows = prev_importance_df[~(prev_importance_df['importance'].isna()) &
+                                                     ~(prev_importance_df.index.isin(using_prev_fit_fi))].sort_values(by='importance')
+        features = unevaluated_rows.index.tolist() + prev_fit_evaluated_rows.index.tolist() + curr_fit_evaluated_rows.index.tolist()
+    if auto_threshold:
+        features = list(prioritized) + [feature for feature in features if feature not in prioritized]
+    return features
+
+
+class FeatureSelector:
+    def __init__(self, model: AbstractModel, time_limit: float, seed: int = 0, keep_models: bool = False, raise_exception=False) -> None:
```

Comment:
why not have proxy model as another (optional) argument

Reply:
also proxy_model should probably be variable name in the code, given it is mentioned numerous time in the comments and in the `from_proxy` args, otherwise is a bit confusing what the proxy model is exactly.